scipy/spatial/ckdtree.pyx

```diff
@@ -503,13 +503,13 @@ cdef public class cKDTree [object ckdtree, type ckdtree_type]:
             self.boxsize_data = None
         else:
             boxsize_arr = np.empty(2 * self.m, dtype=np.float64)
-            boxsize_arr[:] = boxsize
+            boxsize_arr[:self.m] = boxsize
             boxsize_arr[self.m:] = 0.5 * boxsize_arr[:self.m]
             # FIXME: how to use a matching del if new is used?
             self.boxsize_data = boxsize_arr
             self.raw_boxsize_data = <np.float64_t*> np.PyArray_DATA(boxsize_arr)
             self.boxsize = boxsize_arr[:self.m].copy()
-            if (self.data >= self.boxsize[None, :]).any():
+            if (self.data >= self.boxsize[None, self.boxsize > 0]).any():
```

Comment:
Looks like it is implied that dimensions with <=0 boxsizes are neglected? wouldn't it actually be 
```
(self.data >=self.boxsize[None, :])[self.boxsize > 0].any()
```

We may need to keep track of this information (self.boxsize > 0) in other parts of the code too.